src/spyglass/utils/dj_helper_fn.py

```diff
@@ -8,6 +8,35 @@
 from .nwb_helper_fn import get_nwb_file
 
 
+def _delete(tbl: dj.Table, **kwargs) -> None:
+    descendants = tbl.descendants()
+    rows = tbl.fetch()
+    parents = []
+    parts = []
+    for table in descendants:
+        parent = table.split(".")[-1].split("`__")[-1]
+        if len(parent.split("__")) > 1:
+            parts.append(table)
+        else:
+            parents.append(table)
+    parts_no_parents = [
+        table
+        for table in parts
+        if (table.replace(f'__{table.split("__")[-1]}', "") + "`")
+        not in parents
+    ]
+    for part in parts_no_parents:
+        first_part, last_part = part.split(".")
+        last_part = last_part.split("__")[0]
+        table = f"{first_part}.{last_part}`"
+        conn = dj.conn()
```

Comment:
Could also probably move `conn` outside the for loop in order to avoid multiple calls.